writer.go

```diff
@@ -0,0 +1,39 @@
+package ftdc
+
+import (
+	"io"
+
+	"github.com/mongodb/mongo-go-driver/bson"
+	"github.com/pkg/errors"
+)
+
+type writerCollector struct {
+	writer    io.WriteCloser
+	collector *streamingDynamicCollector
+}
+
+func NewWriterCollector(chunkSize int, writer io.WriteCloser) io.WriteCloser {
+	return &writerCollector{
+		writer: writer,
+		collector: &streamingDynamicCollector{
```

Comment:
Why not use `NewStreamingDynamicCollector` here? https://github.com/mongodb/ftdc/blob/82de4a82e6d62196f8f1fed6c883fd738b0fada0/collector_streaming.go#L91

Reply:
becasue it returns a Collector type and I wanted it to be of the concrete type.

Reply:
ack